drivers/bluetooth/hci/userchan.c

```diff
@@ -114,48 +115,53 @@ static uint16_t packet_len(const uint8_t *buf)
 
 		/* Parameter Total Length */
 		payload_len = cmd->param_len;
-		header_len = BT_HCI_CMD_HDR_SIZE;
+		header_len = BT_HCI_CMD_HDR_SIZE + sizeof(type);
 		break;
 	}
 	case H4_ACL: {
 		const struct bt_hci_acl_hdr *acl = (const struct bt_hci_acl_hdr *)hdr;
 
 		/* Data Total Length */
 		payload_len = sys_le16_to_cpu(acl->len);
-		header_len = BT_HCI_ACL_HDR_SIZE;
+		header_len = BT_HCI_ACL_HDR_SIZE + sizeof(type);
 		break;
 	}
 	case H4_SCO: {
 		const struct bt_hci_sco_hdr *sco = (const struct bt_hci_sco_hdr *)hdr;
 
 		/* Data_Total_Length */
 		payload_len = sco->len;
-		header_len = BT_HCI_SCO_HDR_SIZE;
+		header_len = BT_HCI_SCO_HDR_SIZE + sizeof(type);
 		break;
 	}
 	case H4_EVT: {
 		const struct bt_hci_evt_hdr *evt = (const struct bt_hci_evt_hdr *)hdr;
 
 		/* Parameter Total Length */
 		payload_len = evt->len;
-		header_len = BT_HCI_EVT_HDR_SIZE;
+		header_len = BT_HCI_EVT_HDR_SIZE + sizeof(type);
 		break;
 	}
 	case H4_ISO: {
 		const struct bt_hci_iso_hdr *iso = (const struct bt_hci_iso_hdr *)hdr;
 
 		/* ISO_Data_Load_Length parameter */
 		payload_len =  bt_iso_hdr_len(sys_le16_to_cpu(iso->len));
-		header_len = BT_HCI_ISO_HDR_SIZE;
+		header_len = BT_HCI_ISO_HDR_SIZE + sizeof(type);
```

Comment:
This might just be a minor detail but you are repeating the addition of the length of the H4 packet type (i.e., `sizeof(type)`) in all of the cases.  Would it not be better to make the addition after the switch case? It will not affect the logic but improves readability and removes code repetition.